rpc/lighthouse.go

```diff
@@ -77,6 +58,14 @@ func (lc *LighthouseClient) GetNewBlockChan() chan *types.Block {
 		}
 		defer stream.Close()
 
+		go func() {
+			for {
+				// It is important to register to Errors, otherwise the stream does not reconnect if the connection was lost
+				err := <-stream.Errors
+				utils.LogError(err, "Lighthouse connection error (will automaticaly retry to connect)", 0)
+			}
+		}()
```

Comment:
**Suggestion:**
This does not require another Go routine. Why not add this in the for below like this:
```
for {
	select {
	case err := <-stream.Errors:
		// It is important to register to Errors, otherwise the stream does not reconnect if the connection was lost
		utils.LogError(err, "Lighthouse connection error (will automatically retry to connect)", 0)
	case e := <-stream.Events:
		var parsed StreamedBlockEventData
		err = json.Unmarshal([]byte(e.Data()), &parsed)
		...
```

Reply:
sure